test/controllers/trips_controller_test.rb

```diff
@@ -1,23 +1,266 @@
+      update_trip = Trip.find_by(id: id)
+      expect(update_trip.driver_id).must_equal @trip.driver_id
+      expect(update_trip.passenger_id).must_equal @trip.passenger_id
+      expect(update_trip.date).must_equal valid_trip[:trip][:date]
+      expect(update_trip.rating).must_equal valid_trip[:trip][:rating]
+      expect(update_trip.cost).must_equal valid_trip[:trip][:cost]
+
+      # Check that the controller redirected the user
+      must_redirect_to trip_path(id)
+    end
+
+    it "does not update any driver if given an invalid id, and responds with a 404" do
+      # Arrange
+      id = -1
+
+      # Act-Assert
+      expect {
+        patch trip_path(id), params: valid_trip
+      }.wont_change 'Trip.count'
+
+      # Assert
+      must_respond_with :not_found
+    end
+
+    it "does not update a trip if the form data violates rating, and responds with a redirect" do
+      # Arrange
+      id = @trip.id
+
+      # Act-Assert
+      expect {
+        patch trip_path(id), params: invalid_trip
+      }.wont_change 'Trip.count'
+
+      # Assert
+      must_respond_with :bad_request
+    end
   end
 
   describe "destroy" do
-    # Your tests go here
+    before do
+      @driver = Driver.create(name: "TEST123", vin: "WBWSS52P9NEYLVDE9", available: true, isactive: true)
+      @passenger = Passenger.create(name: "Judy", phone_num: "360-555-0987")
+      @trip = Trip.create(driver_id: @driver.id, passenger_id: @passenger.id, date: "2016-04-05", rating: 3, cost: 1293)
+    end
+
+    it "can delete the trip instance in db when trip exists and both driver and passenger are inactive, then redirects" do
+      # Arrange
+      valid_trip = @trip.id
+      # inactive driver & passenger
+      @driver.update(available: false, isactive: false)
+      @passenger.update(isactive: false)
+      
+      # Act-Assert
+      expect {
+        delete trip_path(valid_trip)
+      }.must_change "Trip.count", 1
+      
+      expect(@driver.isactive).must_equal false
+      expect(@passenger.isactive).must_equal false
+
+      # Assert
+      must_redirect_to root_path
+    end
+
+    it "does not change the db when the trip does not exist, then responds with 404" do
+      # Arrange
+      id = -1
+
+      # Act-Assert
+      expect {
+        delete trip_path(id)
+      }.wont_change "Trip.count"
+
+      # Assert
+      must_respond_with :not_found
+    end
+
+    it "does not change the db when trip exists but either driver and passenger are still active, then redirects" do
+      # Arrange
+      valid_trip = @trip.id
+      # inactive driver & passenger
+      @driver.update(available: false, isactive: false)
+
+      # Act-Assert
+      expect {
+        delete trip_path(valid_trip)
+      }.wont_change "Trip.count"
+      
+      expect(@driver.isactive).must_equal false
+      expect(@passenger.isactive).must_equal true
+
+      # Assert
+      must_redirect_to trip_path(@trip.id)
+    end
+  end
+
+  describe "complete_trip" do
```

Comment:
I would also include a test to verify that the driver's average rating changed appropriately.